### ISSUE TITLE
ComVisible break file operation on Mac OS X

### ISSUE THREAD

**Issue by cnblogs-dudu** [dotnet/corefx#779]
Run the following very simple console app that creates a file on Mac OS X by corerun.
```charp
using System;

internal class Program
{
    private static void Main(string[] args)
    {        
        var filePath  = "/git/dotnet/CoreClrHelloWorld/HelloWorld.log";
        Console.WriteLine("Creating " + filePath + "...");
        System.IO.File.Create(filePath);
        Console.WriteLine("Created " + filePath);
        Console.WriteLine("Press ENTER to exit ...");
        Console.ReadLine();
    }
}
```
But result in the following failures.
```
Assert failure (unable to format)
/git/dotnet/coreclr/src/utilcode/ccomprc.cpp
false && "UNIXTODO: Implement string loading from resources"
**** MessageBox invoked, title 'Assert failure (unable to format)' ****
  false && "UNIXTODO: Implement string loading from resources"
********

Assert failure (unable to format)
/git/dotnet/coreclr/src/utilcode/ccomprc.cpp
false && "UNIXTODO: Implement string loading from resources"
**** MessageBox invoked, title 'Assert failure (unable to format)' ****
  false && "UNIXTODO: Implement string loading from resources"
********

Assert failure (unable to format)
/git/dotnet/coreclr/src/utilcode/ccomprc.cpp
false && "UNIXTODO: Implement string loading from resources"
**** MessageBox invoked, title 'Assert failure (unable to format)' ****
  false && "UNIXTODO: Implement string loading from resources"
********

{0x7fff761bf300-0x10c908de0} ASSERT [DEBUG  ] at 
/git/dotnet/coreclr/src/pal/src/arch/i386/context.cpp.38: Trace/BPT trap: 5
```
So no file has been created.
After that ... While I comment `[System.Runtime.InteropServices.ComVisible(true)]` in FileAccess.cs and FileMode.cs and FileShare.cs in System.IO.FileSystem.Primitives, the file was created successfully.

**Comment by stephentoub** [comment]
@jkotas, do you know why this would fail like this?  

Regardless, beyond any changes that may be needed in coreclr, should we be stripping all ComVisible attributes from all of the corefx libraries, since they're not needed with .NET Core, right?  There are currently ~130 of them in the GitHub repo.

**Comment by richlander** [comment]
@stephentoub ComVisible really only makes sense if we're providing an in-proc server for .NET Core, which we're not. That would push us back down the mscoree.dll activation path, since COM type registrations (by default) are global. I'm sure that there is a way to make this all work with reg-free app-local fusion-style registrations, but that seems like an even worse idea.

I assume that sharing changes back to the .NET Framework BCL from corefx using the copy/paste method. Will removing these attributes make that engineering tasks harder? I assume not. It's not like we're copying whole types.

**Comment by jkotas** [comment]
> do you know why this would fail like this?

Methods that read ComVisible attribute in vm\interoputil.cpp are not under FEATURE_COMINTEROP, and I guess there is a problem on that path somewhere. coreclr side of the fix should include placing the ComVisible attribute reading under FEATURE_COMINTEROP.

> Should we be stripping all ComVisible attributes from all of the corefx libraries, since they're not needed with .NET Core?

Yes. I believe it has been discussed a couple of times but nobody have done it yet...

**Comment by stephentoub** [comment]
> Yes. I believe it has been discussed a couple of times but nobody have done it yet...

I'll do it.

**Comment by stephentoub** [comment]
Fixed by #782

**Comment by japj** [comment]
Is #276 still something that needs to be done as part of this, or is that not needed anymore?
